editor/editor_asset_installer.cpp

```diff
@@ -82,229 +81,259 @@ void EditorAssetInstaller::open(const String &p_path, int p_depth) {
 		char fname[16384];
 		unzGetCurrentFileInfo(pkg, &info, fname, 16384, nullptr, 0, nullptr, 0);
 
-		String name = String::utf8(fname);
-		files_sorted.insert(name);
+		String source_name = String::utf8(fname);
 
-		ret = unzGoToNextFile(pkg);
-	}
+		// Create intermediate directories if they aren't reported by unzip.
+		// We are only interested in subfolders, so skip the root slash.
+		int separator = source_name.find("/", 1);
+		while (separator != -1) {
+			String dir_name = source_name.substr(0, separator) + "/";
```

Comment:
Nit: Could likely use `split("/")` instead of manual `find` and `substr`, but it's not important.

Reply:
Or better, `get_slice()` and `get_slice_count()`.

Reply:
It's actually better that way in this case. If anything, it could be replaced with `get_slice()`.

Reply:
Not sure that split would be an improvement, we would still need to iterate over the collection. `get_slice`/`get_slice_count` also seem to be less efficient for this, I think. But I agree with the final suggestion, it makes sense.